flutter/lib/ui/geometry.dart

```diff
@@ -680,35 +671,102 @@ class Rect {
   String toString() => "Rect.fromLTRB(${left.toStringAsFixed(1)}, ${top.toStringAsFixed(1)}, ${right.toStringAsFixed(1)}, ${bottom.toStringAsFixed(1)})";
 }
 
-// A rounded rectangle with the same radii for all four corners.
+/// A radius for either circular or oval shapes.
+class Radius {
+  /// Constructs a circular radius.
+  const Radius.circular(double radius) : this.oval(radius, radius);
+
+  /// Construct an oval radius.
+  const Radius.oval(double this.x, double this.y);
+
+  /// The radius value on the horizontal x axis.
+  final double x;
+
+  /// The radius value on the vertical y axis.
+  final double y;
+}
+
+/// A rounded rectangle with the custom radii for all four corners.
 class RRect {
   RRect._();
 
   /// Construct a rounded rectangle from its left, top, right, and bottom edges,
-  /// and the radii along its horizontal axis and its vertical axis.
-  RRect.fromLTRBXY(double left, double top, double right, double bottom, double radiusX, double radiusY) {
+  /// and a radius that is constant in each corner.
+  RRect.fromLTRBConst(double left, double top, double right, double bottom,
```

Comment:
actually, I would combine this one with `fromLTRBCustom`, with the last three arguments being optional, and with the convention being that if the topRight is omitted, it defaults to the value of topLeft, if the bottomRight is omitted it defaults to the value of topRight, and if the bottomLeft is omitted, it defaults to the value of topLeft.

Maybe even make the four arguments named argument rather than positional arguments.